core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala

```diff
@@ -106,6 +106,13 @@ private[spark] abstract class BasePythonRunner[IN, OUT](
     val startTime = System.currentTimeMillis
     val env = SparkEnv.get
     val localdir = env.blockManager.diskBlockManager.localDirs.map(f => f.getPath()).mkString(",")
+    // if OMP_NUM_THREADS is not explicitly set, override it with the number of cores
+    if (conf.getOption("spark.executorEnv.OMP_NUM_THREADS").isEmpty) {
+      // SPARK-28843: limit the OpenMP thread pool to the number of cores assigned to this executor
+      // this avoids high memory consumption with pandas/numpy because of a large OpenMP thread pool
+      // see https://github.com/numpy/numpy/issues/10455
```

Comment:
Can we avoid to say it targets pandas/numpy? This PR I guess then targets to fix a general issue rather than one specific issue found in numpy.

Reply:
Can we clarify here? I think we don't usually open a PR to make such change in followups.

What does "limit the OpenMP thread pool to the number of cores assigned to this executor" means in general PySpark jobs?

Also, can you please clarify this in migration guide?

Reply:
Can we clarify here? What does "limit the OpenMP thread pool to the number of cores assigned to this executor" means in general PySpark jobs?